File: examples/documents/model-update.swift
import JsBaoClient

// Update a record: load it, mutate fields, save.
func completeTask(documentId: String, taskId: String) async throws {
  // #region example
  if var task = try await Task.find(taskId) {
    task.completed = true
    try task.save(in: documentId)
  }
  // #endregion example
}

File: examples/documents/model-update.ts
import { Task } from "../_harness/generated/ts/Task.generated";

// Update a record: load it, mutate fields, save.
export async function completeTask(taskId: string) {
  // #region example
  const task = await Task.find(taskId);
  if (task) {
    task.completed = true;
    await task.save();
  }
  // #endregion example
}
